### PR TITLE
Run `cargo fmt` on entire codebase

### PR DESCRIPTION
**Pull request by derekdreery**
I'm basically making this pull request to plead for you to run `cargo fmt` on the codebase so I don't have to spend hours unpicking all the changes in #930, but it would also be good from the point of view of consistency. `rustfmt` is stable now with semver guarantees.

I included an empty rustfmt.toml file to indicate the project uses default settings.

### CODE REVIEW

**Comment by jebrosen** [comment]
Rocket actually had a `rustfmt.toml` file in the past, but it was removed in f171dc9. I think we should reevaluate the decision to not use rustfmt, but let's at least use the previous style as a starting point to reduce the churn if possible.

**Comment by derekdreery** [comment]
I've tried to match the options to the ones that were there in https://github.com/SergioBenitez/Rocket/commit/f171dc9d095b9b1f63ba5bb69fbbbdf4aef7ccf2.

File: contrib/codegen/build.rs [hunk 1]
```diff
@@ -18,38 +18,56 @@ fn main() {
...
+            "Installed version is:",
+            Yellow.paint(format!("{} ({})", version, date)),
+            "Minimum required:",
+            Yellow.paint(format!("{} ({})", MIN_VERSION, MIN_DATE))
+        );
     };
 
-    if let (Some(ok_channel), Some((ok_version, version)), Some((ok_date, date))) = triple {
+    if let (Some(ok_channel), Some((ok_version, version)), Some((ok_date, date))) = triple
+    {
```
[the remark this answers is not in the document]
Comment (derekdreery) [reply contrib/codegen/build.rs]: I think rustfmt will override the AlwaysSameLine behaviour for length.

**Comment by SergioBenitez** [comment]
As much as I'd like to use `rustfmt`, it doesn't do the right thing, even according to the style guidelines that are supposed to be enforced by the tool. Said another way, it's simply not mature enough. Happy to revisit this when it is.

**Comment by jhpratt** [comment]
While many of the options are still unstable, it looks like rustfmt has gained quite of functionality. `inline_attribute_width` handles things like `#[cfg(feature = "alloc")] use core::slice;` and is relatively new (I don't remember seeing it last I checked).

After the async branch is merged into master, I'm going to take a look to see what combination of config settings minimize the line diff, as that would indicate what Rocket is using already. If it seems sensible, I'll continue a discussion on IRC/Matrix and create a new PR.

**Comment by derekdreery** [comment]
OT, but I've now turned off my auto formatting, because I've been bitten by this a few times.

**Pull request closed without merging**